### PR TITLE
Fix Composer staging dockerimages

### PR DESCRIPTION
**Pull request by ravi-mosaicml**
#1197, which introduced composer image staging builds, installs Composer from pypi rather than from github. This PR fixes that by ensuring the modified buildArgs are passed to kaniko correctly.

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -311,12 +300,8 @@ stage('Prepare') {
         echo "gitBranch: $gitBranch"
         echo "gitCommit: $gitCommit"
 
-        didDockerChange = isPathModified('docker/')
-
-        String dockerfile = 'Dockerfile'
-        String buildContext = './docker'
-        String buildMatrix = './docker/build_matrix.yaml'
-        pytorchDockerBuildMatrix = getDockerBuildMatrix(buildMatrix, buildContext, dockerfile)
+        didDockerChange = true  // TODO(Ravi) Unset this hardcode. isPathModified('docker/')
```
Comment (ravi-mosaicml) [inline .ci/Jenkinsfile:303]: TODO

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -355,7 +331,7 @@ stage('Build') {
             }
             if (!isMergeCommit) {
                 // Skip composer images except on merge commits -- it's slow!
-                return
+                // return // TODO(ravi) Uncomment this line
```
Comment (ravi-mosaicml) [inline .ci/Jenkinsfile:334]: TODO

File: .ci/Jenkinsfile (hunk 1 above)
Comment (ravi-mosaicml) [reply .ci/Jenkinsfile]: removed

File: .ci/Jenkinsfile (hunk 2 above)
Comment (ravi-mosaicml) [reply .ci/Jenkinsfile:332]:
```suggestion
                return
```

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -330,45 +313,38 @@ stage('Build') {
...
-                // If this is a composer image and it is the latest tag, skip it -- we'll build the versioned tag instead
-                return
-            }
+        if (isComposerImage && buildArgs['COMPOSER_INSTALL_COMMAND'] == 'mosaicml[all]GIT_COMMIT') {
+            // If this is a composer image and it is the 'GIT_COMMIT' entry in the matrix
+            // then build and publish the image on the staging repo
             if (!isMergeCommit) {
                 // Skip composer images except on merge commits -- it's slow!
-                return
+                // return  // TODO (ravi) -- uncomment this line
```
Comment (ravi-mosaicml) [inline .ci/Jenkinsfile:332]: TODO
```suggestion
                return
```

File: docker/generate_build_matrix.py [hunk 4]
```diff
@@ -153,7 +156,8 @@ def _main():
 
     composer_entries = []
 
-    composer_versions = ['', '==0.7.1']  # Only build images for the latest composer version
+    # The `GIT_COMMIT` is a placeholder and will be substituted with the actual git commit
```
Comment (ravi-mosaicml) [inline docker/generate_build_matrix.py:159]:
```suggestion
    # The `GIT_COMMIT` is a placeholder and Jenkins will substitute it with the actual git commit for the `composer_staging` images
```

File: .ci/Jenkinsfile (hunk 3 above)
Comment (bandish-shah) [reply .ci/Jenkinsfile:330]: I'm going to approve to unblock you in the morning but please resolve this before merging.

**Review by bandish-shah — APPROVED** [review]
LGTM, please resolve the TODO's before merging.

File: .ci/Jenkinsfile (hunk 3 above)
Comment (ravi-mosaicml) [reply .ci/Jenkinsfile]: Removed this TODO

**Pull request merged**